### PR TITLE
Tarik/oc 27 quiz question methods

### PR DESCRIPTION
**Pull request by TarikDemo**

### CODE REVIEW

File: src/main/java/com/quiz/service/impl/QuizServiceImpl.java [hunk 1]
```diff
@@ -1,6 +1,129 @@
...
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.util.UUID;
+
 public class QuizServiceImpl implements QuizService {
+    @Override
+    public void addQuizQuestion(Long quizId, String question) {
```
Comment (eycwave) [inline src/main/java/com/quiz/service/impl/QuizServiceImpl.java]: why did you write this in ServiceImpl class? isn't it exist in QuizQuestionRepository? can you explain thank you

File: src/main/java/com/quiz/service/QuizService.java [hunk 2]
```diff
@@ -1,4 +1,18 @@
 package com.quiz.service;
 
+import com.quiz.model.QuizQuestion;
+
 public interface QuizService {
+   void addQuizQuestion(Long quizId,String question);
```
Comment (JamshidS) [inline src/main/java/com/quiz/service/QuizService.java]: return `String`

File: src/main/java/com/quiz/service/QuizService.java [hunk 3]
```diff
@@ -1,4 +1,18 @@
...
+
+   QuizQuestion updateQuizQuestion(Long quizId,Long questionId,String newQuestion);
+
+   void deleteQuizQuestion(Long quizID,Long questionId);
+
+   QuizQuestion getQuizQuestionByQuizId(Long listQuizId);
+
+
```
Comment (JamshidS) [inline src/main/java/com/quiz/service/QuizService.java:17]: create a new method that returns a  `List<QuizQuestion>`  with quizId;

File: src/main/java/com/quiz/service/QuizService.java [hunk 4]
```diff
@@ -1,4 +1,18 @@
...
+import com.quiz.model.QuizQuestion;
+
 public interface QuizService {
+   void addQuizQuestion(Long quizId,String question);
+
+   QuizQuestion updateQuizQuestion(Long quizId,Long questionId,String newQuestion);
+
+   void deleteQuizQuestion(Long quizID,Long questionId);
+
+   QuizQuestion getQuizQuestionByQuizId(Long listQuizId);
```
Comment (JamshidS) [inline src/main/java/com/quiz/service/QuizService.java]:
```suggestion
   QuizQuestion getQuizQuestionById(Long questionId);
```

File: src/main/java/com/quiz/service/impl/QuizServiceImpl.java [hunk 5]
```diff
@@ -1,6 +1,129 @@
...
+
+
+        String query = "INSERT INTO quizquestion (uuid, quizId, question, createdDateTime) VALUES (?, ?, ?, ?)";
```
Comment (JamshidS) [inline src/main/java/com/quiz/service/impl/QuizServiceImpl.java]: this is wrong just remove it.

File: src/main/java/com/quiz/service/impl/QuizServiceImpl.java [hunk 6]
```diff
@@ -1,6 +1,129 @@
...
 
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.util.UUID;
+
 public class QuizServiceImpl implements QuizService {
+    @Override
```
Comment (JamshidS) [inline src/main/java/com/quiz/service/impl/QuizServiceImpl.java]: private final QuizQuestionRepository quizQuestionRepository;
// constructor injection. -> dependency injection

**Review by JamshidS — CHANGES_REQUESTED** [review]

File: src/main/java/com/quiz/repository/QuizQuestionRepository.java [hunk 7]
```diff
@@ -111,29 +112,60 @@ public QuizQuestion getQuizQuestionByID(int id){
...
                 }
+            }
         } catch (SQLException e) {
             e.printStackTrace();
         }
         return quizQuestion;
     }
 
-    public List<QuizQuestion> getAllQuizQuestion(){
+    public List<QuizQuestion> getAllQuizQuestion() {
```
Comment (JamshidS) [inline src/main/java/com/quiz/repository/QuizQuestionRepository.java]:
```suggestion
    public List<QuizQuestion> getAllQuizQuestions() {
```

File: src/main/java/com/quiz/service/impl/QuizServiceImpl.java [hunk 8]
```diff
@@ -1,6 +1,78 @@
...
+           quizQuestionRepository.deleteQuizQuestion(questionId);
+           return "Quiz deleted successfully.";
+       }
+
+    }
+
+
+    @Override
+    public List<QuizQuestion> getAllQuizQuestion() {
+        return quizQuestionRepository.getAllQuizQuestion();
```
Comment (JamshidS) [inline src/main/java/com/quiz/service/impl/QuizServiceImpl.java]:
```suggestion
        return quizQuestionRepository.getAllQuizQuestions();
```

**Review by JamshidS — CHANGES_REQUESTED** [review]

**Review by JamshidS — APPROVED** [review]

**Pull request merged**